app/models/katello/cdn_configuration.rb

```diff
@@ -1,28 +1,57 @@
 module Katello
   class CdnConfiguration < Katello::Model
     include Encryptable
+    self.inheritance_column = nil
+    CDN_TYPE =  'redhat_cdn'.freeze
+    UPSTREAM_SERVER_TYPE =  'upstream_server'.freeze
+    AIRGAPPED_TYPE = 'airgapped'.freeze
+
+    TYPES = [CDN_TYPE, UPSTREAM_SERVER_TYPE, AIRGAPPED_TYPE].freeze
 
     belongs_to :organization, :inverse_of => :cdn_configuration
 
     belongs_to :ssl_ca_credential, :class_name => "Katello::ContentCredential", :inverse_of => :ssl_ca_cdn_configurations
 
     encrypts :password
 
-    validates :url, presence: true
-    validates_with Validators::KatelloUrlFormatValidator, attributes: :url
+    validates :url, presence: true, unless: :airgapped?
+    validates_with Validators::KatelloUrlFormatValidator, attributes: :url, unless: :airgapped?
     validates_with Validators::KatelloLabelFormatValidator, attributes: :upstream_organization_label, if: proc { upstream_organization_label.present? }
-    validate :non_redhat_configuration, unless: :redhat?
+    validate :non_redhat_configuration, if: :upstream_server?
+
+    before_validation :reset_fields
 
     def ssl_ca
       ssl_ca_credential&.content
     end
 
-    def redhat?
-      username.blank? && password.blank? && upstream_organization_label.blank? && ssl_ca_credential_id.blank?
+    def cdn?
```

Comment:
Apologies if I missed a prior discussion - why call this type just 'cdn?' - it's a bit nonspecific. 'Why not redhat'?

Reply:
I dont have a preference. But I thought calling it CDN made it clearer ? Open for name suggestions.

Reply:
It checks for cdn type

Reply:
it's called 'red hat' in the UI though so I think it's nice the the type matches that designation

Reply:
I can rename that helper method to `redhat_cdn`

Reply:
Done:)